src/guest_memory.rs

```diff
@@ -165,6 +174,9 @@ impl FileOffset {
 /// Represents a continuous region of guest physical memory.
 #[allow(clippy::len_without_is_empty)]
 pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
+    /// Type used for dirty memory tracking.
```

Comment:
In case we may add more associated types in future, how about:
```
trait GuestMemoryTypes {
       type B: Bitmap;
}

pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     type T: GuestMemoryTypes;
}
```
It's much easier to add more associated types in future.